render-test/allocation_index.cpp

```diff
@@ -0,0 +1,93 @@
+#include "allocation_index.hpp"
+
+#include <atomic>
+#include <cassert>
+#include <cstdlib>
+#include <mutex>
+#include <unordered_map>
+
+namespace {
+
+std::atomic_size_t indexedMemorySize{0};
```

Comment:
These are already protected by mutex, is atomic needed here?

Reply:
I just did not want to lock mutex on stats getters (e.g. `AllocationIndex::getAllocatedSize()`), would you prefer the other way?

Reply:
no, it is fine, thanks for explanation.